packages/core-flows/src/definition/cart/steps/refresh-cart-shipping-methods.ts

```diff
@@ -0,0 +1,89 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { CartDTO, ICartModuleService } from "@medusajs/types"
+import { arrayDifference } from "@medusajs/utils"
+import { createStep, StepResponse } from "@medusajs/workflows-sdk"
+import { IFulfillmentModuleService } from "../../../../../types/dist/fulfillment/service"
+
+interface StepInput {
+  cart: CartDTO
+}
+
+export const refreshCartShippingMethodsStepId = "refresh-cart-shipping-methods"
+export const refreshCartShippingMethodsStep = createStep(
+  refreshCartShippingMethodsStepId,
+  async (data: StepInput, { container }) => {
+    const { cart } = data
+    const { shipping_methods: shippingMethods = [] } = cart
+
+    if (!shippingMethods?.length) {
+      return new StepResponse(void 0)
+    }
+
+    const fulfillmentModule = container.resolve<IFulfillmentModuleService>(
+      ModuleRegistrationName.FULFILLMENT
+    )
+
+    const cartModule = container.resolve<ICartModuleService>(
+      ModuleRegistrationName.CART
+    )
+
+    const shippingMethodsToDelete: string[] = []
+    const shippingOptionIds: string[] = shippingMethods.map(
+      (sm) => sm.shipping_option_id!
+    )
+
+    const shippingOptions = await fulfillmentModule.listShippingOptions(
+      { id: shippingOptionIds },
+      { select: ["id", "name"], take: null }
+    )
+
+    const diff = arrayDifference<string>(
```

Comment:
Let's just make sure the difference is the methods that were in the cart but not in the DB, not the way around (it might be correct, just wanted to mention it as something that can easily fall through the cracks)